### PR TITLE
fix and improve overclocking logic

### PR DESCRIPTION
**Pull request by TechLord22**
## What
This PR fixes issues with overclocking logic, causing recipes to often overclock one time fewer than they were supposed to.

## Implementation Details
In making this PR, the PA's custom overclocking logic was found to be largely redundant and was now removed.

## Outcome
Fixes problems with overclocking logic.

### CODE REVIEW

**Review by Syrcan — APPROVED** [review]
LGTM

File: src/test/java/gregtech/api/recipes/logic/OverclockingTest.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        // 0 overclocks
+        assertEquals(recipeVoltage, oc[0]);
+        assertEquals(recipeDuration, oc[1]);
+
+        // ULV recipe, MV machine
+        machineTier = MV;
+
+        oc = testOC(recipeDuration, recipeTier, recipeVoltage, machineTier, (int) V[machineTier]);
+
+        // 0 overclocks
```
Comment (ALongStringOfNumbers) [inline src/test/java/gregtech/api/recipes/logic/OverclockingTest.java:30]: This comment needs to be adjusted, probably just copy-paste error

File: src/test/java/gregtech/api/recipes/logic/OverclockingTest.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        assertEquals(recipeVoltage * 4, oc[0]);
+        assertEquals(recipeDuration / 2, oc[1]);
+
+        // ULV recipe, HV machine
+        machineTier = HV;
+
+        oc = testOC(recipeDuration, recipeTier, recipeVoltage, machineTier, (int) V[machineTier]);
+
+        // 0 overclocks
```
Comment (ALongStringOfNumbers) [inline src/test/java/gregtech/api/recipes/logic/OverclockingTest.java:39]: This comment needs to be adjusted, probably just copy-paste error

**Review by ALongStringOfNumbers — CHANGES_REQUESTED** [review]
Needs tests for PA, which has to account for the machines in the machine hatch, and the tier of energy the multiblock is being fed.

Needs tests for the fusion reactor, to ensure that we are always properly enforcing a 2/2 overclock.

This PR does not work properly with the Processing Array. Test Recipe:
Raw Rubber + Concrete -> Construction Foam. Supplied Machines MV mixer. Supplied power 2A UHV. The Processing Array does not start. With some light debugging, this is breaking with parallel recipes.

**Pull request merged**